### PR TITLE
Missing acute in french in self harm2084

### LINKED ISSUE #2084: Missing acute accent in self harm word section
## Summary

1. Missing acute accent in self harm word section (example: Médicaments instead of  Medicaments)
2. Should we use the same word in french for self harm. Currently, we use Auto-nuisibles and automutilation for self harm

## Steps to reproduce

1. Submit a french report which has a self harm words

harm

### PR DESCRIPTION
**Pull request by dengcn**
Fixes #2084 

This is the fix to PR2084 Missing acute accent in self harm word section

# Description

There are 2 issues in this PR
1) In the French alarm warning, acute or accented sign were missing
2) French Flagged field(for alarm) does not match with the title of French alarm warming
  
> Please include a summary of the change.
-In the handling of selfAlarmWordsScan
-In the Language json

Note: In the fix, 
The accute accented sign will be preserved in report,  also if user miss any accute accented sign in french word. it will also be captured.

For example: Déprimé is defined in the env
If user type in Déprimé ===> déprimé in report
If user type in déprimé ===> déprimé in report 
If user type in deprimé ===> déprimé in report 
If user type in deprime ===> déprimé in report 

# Any new packages installed?
No
> Give details

# Required followup work

> Is there anything related to this that still needs to be done (ex: documentation changes).

# Checklist:

- [ ] I have updated the azurescript.sh with any **new environment variables** and added them to the appsettings
- [ x] I have looked at my code on GitHub and it all looks good (ex: no random commented out code or console.logs)
- [ ] I have added and needed tests for my changes (in particular for new screens)
- [ ] I have added a comment to any confusing code
- [ ] I have added documentation to any modified front-end code. (Or added missing documentation)

### CODE REVIEW

File: f2/src/locales/fr.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
   "analystReport.reportLanguage": "Langue du rapport",
   "analystReport.reportNumber": "Signaler le numéro",
   "analystReport.reportVersion": "Version du rapport",
-  "analystReport.selfHarmString": "mots d'automutilation détectés",
```
Comment (justinr86) [inline f2/src/locales/fr.json:17]: Where did this translation come from?

Comment (dengcn) [reply f2/src/locales/fr.json:17]: Coming from Harm Word in title already defined and used, and PR raiser wish this the same as title

File: f2/src/utils/selfHarmWordsScan.js [hunk 2]
```diff
@@ -47,24 +45,36 @@ const selfHarmWordsScan = (data) => {
 //Scan String for key words. Tokenize and stem to identify root words.
 const scanString = (str) => {
   try {
-    let modifiedStr = unidecode(str.toLowerCase())
+    let modifiedStr = str
```
Comment (justinr86) [inline f2/src/utils/selfHarmWordsScan.js:48]: Why declare a variable that has the same value as the param?

Comment (dengcn) [reply f2/src/utils/selfHarmWordsScan.js:48]: Keep the original copy of data, since modifiedStr will be heavily changed or regrouped in the following code

Comment (dengcn) [reply f2/src/utils/selfHarmWordsScan.js:48]: I can consolidate these 2 variables, any others?

**Comment by justinr86** [comment]
From description:

Note: In the fix,
The accute accented sign will be preserved in report, also if user miss any accute accented sign in french word. it will also be captured.

For example: Déprimé is defined in the env
If user type in Déprimé ===> déprimé in report
If user type in déprimé ===> déprimé in report
If user type in deprimé ===> déprimé in report
If user type in deprime ===> déprimé in report

---
Is this desired outcome? The goal of this code change is to maintain the user's input not correct it. 

Also, there seems to be a problem with the report. Though self harm keywords are flagged they are not displayed in the report and there seems to be an encoding issue (not sure if this is related to this change or not), see attached. I submitted a report with the word deprime in the What Happened section.

frenchSubmission

**Comment by justinr86** [comment]
Sorry, I think I misunderstood part of the description. However, there are several failed unit test related to self harm word scan.

**Comment by dengcn** [comment]
Nice catch on this screen (word missing from warning yellow part) from this single harmful word, there was a bug associated with handling such single word, I will fix that.

**Comment by justinr86** [comment]
The associated issue for this PR has been closed. Should this PR be closed as well?

**Comment by dengcn** [comment]
There is an error in the code which needs to be fixed, the pointed PR is still valid, not closed

**Comment by justinr86** [comment]
> There is an error in the code which needs to be fixed, the pointed PR is still valid, not closed

This pull request was created on Aug. 11. Since we are migrating to Azure at the end of the week it might be a good idea to resolve this issue.

**Comment by dengcn** [comment]
Close this PR for now, reopen when migration to Azure is ready

**Pull request closed without merging**